boards/shared/Inc/App/App_SharedSignal.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <stdbool.h>
+#include <stdint.h>
+
+struct Signal;
+
+/**
+ *
+ * @param is_high A function that can be called to check if the signal is high
+ * @param initial_time_ms The initial time, in milliseconds, used to initialize
+ *                        the internal state of the signal
+ * @param duration_high_ms The duration, in milliseconds, that this signal must
+ *                         be continuously high for before the callback function
+ *                         is called
+ * @param callback A callback function that is called when the signal has been
+ *                 continuously high for a period equal to or greater than
+ *                 `duration_high_ms`
+ */
+struct Signal *App_SharedSignal_Create(
+    bool (*is_high)(void),
+    uint32_t initial_time_ms,
+    uint32_t duration_high_ms,
+    void (*callback)(void));
+
+/**
+ * Deallocate the memory used by the given signal
+ * @signal The signal to deallocate
+ */
+void App_SharedSignal_Destroy(struct Signal *signal);
+
+/**
+ * Get the last time the given signal was low, in milliseconds
+ * @param signal The signal to get the time from
+ * @retur The last time the given signal was low, in milliseconds
+ */
+uint32_t App_SharedSignal_GetLastTimeLowMs(const struct Signal *signal);
+
+/**
+ * Get the last time the given signal was high, in milliseconds
+ * @param signal The signal to get the time from
+ * @retur The last time the given signal was high, in milliseconds
+ */
```

Comment:
In general, I'd suggest changing `was high` and `was low` to `was observed to be high` and `was observed to be low` just to be super clear.

Reply:
Fixed!